### PR TITLE
Adding Cephfs Snapshot and clone related test cases to 4.x

### PR DESCRIPTION
**Pull request by AmarnatReddy**
Adding Cephfs Snapshot and clone related test cases to 4.x

Log File : [http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-7YX0D4](http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-7YX0D4)
We have 1 failure in the log for which we have raised BZ : [https://bugzilla.redhat.com/show_bug.cgi?id=2028416](https://bugzilla.redhat.com/show_bug.cgi?id=2028416)

Signed-off-by: Amarnath K <amk@amk.remote.csb>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: suites/nautilus/cephfs/tier_2_cephfs_test-snapshot-clone.yaml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          ceph_conf_overrides:
+            client:
+              rgw crypt require ssl: false
+              rgw crypt s3 kms encryption keys: testkey-1=YmluCmJvb3N0CmJvb3N0LWJ1aWxkCmNlcGguY29uZgo=
+                testkey-2=aWIKTWFrZWZpbGUKbWFuCm91dApzcmMKVGVzdGluZwo=
+            global:
+              mon_max_pg_per_osd: 1024
+              osd_default_pool_size: 2
+              osd_pool_default_pg_num: 64
+              osd_pool_default_pgp_num: 64
```
Comment (psathyan) [inline suites/nautilus/cephfs/tier_2_cephfs_test-snapshot-clone.yaml:48]: Do you need these overrides?

File: suites/nautilus/cephfs/tier_2_cephfs_test-snapshot-clone.yaml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            mon:
+              mon_allow_pool_delete: true
+          ceph_docker_image: rhceph/rhceph-4-rhel8
+          ceph_docker_image_tag: latest
+          ceph_docker_registry: registry.redhat.io
+          ceph_origin: distro
+          ceph_repository: rhcs
+          ceph_stable: true
+          ceph_stable_release: nautilus
```
Comment (psathyan) [inline suites/nautilus/cephfs/tier_2_cephfs_test-snapshot-clone.yaml:57]: These are specific to upstream and can be removed from our configuration

File: suites/nautilus/cephfs/tier_2_cephfs_test-snapshot-clone.yaml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          ceph_stable_rh_storage: true
+          ceph_test: true
+          cephfs_pools:
+            -
+              name: cephfs_data
+              pgs: "8"
```
Comment (psathyan) [inline suites/nautilus/cephfs/tier_2_cephfs_test-snapshot-clone.yaml:63]: Are you sure you what this configuration?

File: suites/nautilus/cephfs/tier_2_cephfs_test-snapshot-clone.yaml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+          containerized_deployment: true
+          copy_admin_key: true
+          dashboard_admin_password: p@ssw0rd
+          dashboard_admin_user: admin
+          dashboard_enabled: true
+          grafana_admin_password: p@ssw0rd
+          grafana_admin_user: admin
+          grafana_container_image: "registry.redhat.io/rhceph/rhceph-4-dashboard-rhel8:4"
+          journal_size: 1024
```
Comment (psathyan) [inline suites/nautilus/cephfs/tier_2_cephfs_test-snapshot-clone.yaml:75]: If required, could you add in the docstring about the environment specific requirement.

**Comment by AmarnatReddy** [comment]
Updated the suite file.
Latest Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-H0TCK9/

**Pull request merged**